lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -105,15 +105,15 @@ type PutWalletPassphrase = "wallets"
   See also: https://input-output-hk.github.io/cardano-wallet/api/#tag/Transactions
 -------------------------------------------------------------------------------}
 
-type Transactions =
-    CreateTransaction
+type Transactions t =
+    CreateTransaction t
 
 -- | https://input-output-hk.github.io/cardano-wallet/api/#operation/postTransaction
-type CreateTransaction = "wallets"
+type CreateTransaction t = "wallets"
```

Comment:
Does this type constraint mean that the wallet API won't be able to make payments to addresses which are of an encoding different to `t`?

Reply:
That is a valid point and the answer is: yes. This means we'll only parse address according to the encoding supported by `t` (note that, this is already implicitly the case at the moment. If one uses something else than `Base58`, the API returns an error expecting an address in `Base58`). 

Whether we should allow different encoding for addresses is up to discussion I believe, although there's no clear use-case for it at the moment I believe. That's a very valid question to ask ourselves (and product) however.